sklearn/neighbors/graph.py

```diff
@@ -180,5 +186,279 @@ def radius_neighbors_graph(X, radius, mode='connectivity', metric='minkowski',
     else:
         _check_params(X, metric, p, metric_params)
 
-    query = _query_include_self(X, include_self)
+    query = _query_include_self(X._fit_X, include_self, mode)
     return X.radius_neighbors_graph(query, radius, mode)
+
+
+class KNeighborsTransformer(NeighborsBase, KNeighborsMixin,
+                            UnsupervisedMixin, TransformerMixin):
+    """Transform X into a (weighted) graph of k nearest neighbors
+
+    The transformed data is a sparse graph as returned by kneighbors_graph.
+
+    .. versionadded:: 0.21
+
+    Parameters
+    ----------
+    mode : {'distance', 'connectivity'}, optional (default = 'distance')
+        Type of returned matrix: 'connectivity' will return the connectivity
+        matrix with ones and zeros, and 'distance' will return the distances
+        between neighbors according to the given metric.
+
+    n_neighbors : int, optional (default = 5)
+        Number of neighbors to use for :meth:`kneighbors` queries. For
```

Comment:
The reference to kneighbors queries is a bit awkward. We're describing what the output of transform is, yeah?

Should we prescribe this as "the minimum number of nearest neighbors to return for each sample"??? I'm a little bit uncertain about how to properly document the proposed behaviour.

Reply:
I agree it was a bit awkward. What do you think of:
```
       Number of neighbors for each sample in the transformed sparse graph.
       For compatibility reasons, as each sample is considered as its own
       neighbor, one extra neighbor will be computed when mode == 'distance'.
       In this case, the sparse graph contains (n_neighbors + 1) neighbors.
```
I am not sure to understand your suggestion. What do you mean by `the minimum number of nearest neighbors`?